test/tools/ossfuzz/protoToSol.cpp

```diff
@@ -0,0 +1,228 @@
+
+#include <libsolutil/Whiskers.h>
+
+#include <sstream>
+
+using namespace solidity::test::solprotofuzzer;
+using namespace solidity::util;
+using namespace std;
+
+string ProtoConverter::protoToSolidity(Program const& _p)
+{
+	// Create random number generator with fuzzer supplied
+	// seed.
+	m_randomGen = make_shared<SolRandomNumGenerator>(_p.seed());
+	return visit(_p);
+}
+
+pair<string, string> ProtoConverter::generateTestCase(TestContract const& _testContract)
+{
+	ostringstream testCode;
+	string usingLibDecl;
+	switch (_testContract.type())
+	{
+	case TestContract::LIBRARY:
+	{
+		m_libraryTest = true;
+		auto testTuple = pseudoRandomLibraryTest();
+		m_libraryName = get<0>(testTuple);
+		Whiskers u(R"(<ind>using <libraryName> for uint;)");
+		u("ind", "\t");
+		u("libraryName", get<0>(testTuple));
+		usingLibDecl = u.render();
+		Whiskers test(R"(<endl><ind><varDecl><endl><ind><ifStmt>)");
+		test("endl", "\n");
+		test("ind", "\t\t");
+		test("varDecl", "uint x;");
+		Whiskers ifStmt(R"(if (<cond>)<endl><ind>return 1;)");
+		Whiskers ifCond(R"(x.<testFunction>() != <expectedOutput>)");
+		ifCond("testFunction", get<1>(testTuple));
+		ifCond("expectedOutput", get<2>(testTuple));
+		ifStmt("cond", ifCond.render());
+		ifStmt("endl", "\n");
+		ifStmt("ind", "\t\t\t");
+		test("ifStmt", ifStmt.render());
+		break;
+	}
+	case TestContract::CONTRACT:
+	{
+		unsigned errorCode = 1;
+		unsigned contractVarIndex = 0;
+		for (auto const& testTuple: m_contractTests)
+		{
+			// Do this to avoid stack too deep errors
+			// We require uint as a return var, so we
+			// cannot have more than 16 variables without
+			// running into stack too deep errors
+			if (contractVarIndex >= s_maxVars)
+				break;
+			string contractName = testTuple.first;
+			string contractVarName = "tc" + to_string(contractVarIndex);
+			Whiskers init(R"(<endl><ind><contractName> <contractVarName> = new <contractName>();)");
+			init("endl", "\n");
+			init("ind", "\t\t");
+			init("contractName", contractName);
+			init("contractVarName", contractVarName);
+			testCode << init.render();
+			for (auto const& t: testTuple.second)
+			{
+				Whiskers tc(R"(<endl><ind><ifStmt>)");
+				tc("endl", "\n");
+				tc("ind", "\t\t");
+				Whiskers ifStmt(R"(if (<cond>)<endl><ind>return <errorCode>;)");
+				Whiskers ifCond(R"(<contractVarName>.<testFunction>() != <expectedOutput>)");
+				ifCond("contractVarName", contractVarName);
+				ifCond("testFunction", t.first);
+				ifCond("expectedOutput", t.second);
+				ifStmt("endl", "\n");
+				ifStmt("cond", ifCond.render());
+				ifStmt("ind", "\t\t\t");
+				ifStmt("errorCode", to_string(errorCode));
+				tc("ifStmt", ifStmt.render());
+				testCode << tc.render();
+				errorCode++;
+			}
+			contractVarIndex++;
+		}
+		break;
+	}
+	}
+	// Expected return value when all tests pass
+	testCode << Whiskers(R"(<endl><ind>return 0;)")("endl", "\n")("ind", "\t\t").render();
+	return pair(usingLibDecl, testCode.str());
```

Comment:
```suggestion
	return {usingLibDecl, testCode.str()};
```
Not too important...